Add documentation for publishing, retrieving keys. (#573)
google/exposure-notifications-server: docs/getting-started/publishing-keys.md

---
layout: default
---

# Key Publishing Guide

This page explains how to configure the server and application to allow
publishing keys.

## Prerequisites

If you haven't started developing a mobile app, please take a look at the
[Android Exposure Notifications Reference App](https://github.com/google/exposure-notifications-android)

You will need to
[complete the necessary steps](https://support.google.com/googleplay/android-developer/contact/expo_notif_api)
to use the exposure notifcations API. You should also ensure that you have
taken the steps to comply with the
[Terms of Service](https://blog.google/documents/72/Exposure_Notifications_Service_Additional_Terms.pdf).
For instance, your app may not request any of the following permissions:
  * ACCESS_FINE_LOCATION
  * ACCESS_COARSE_LOCATION
  * ACCESS_BACKGROUND_LOCATION
  * BLUETOOTH_ADMIN

> ## Publishing Keys

Reviewer: we should say something about region here.

it's rather undefined as far as we are concerned- the server operator needs to agree upon region designations w/ the app and public heath authority

In order for your application to publish keys to the server, the server
requires the addition of the Application Name (for Android) and the Bundle ID
(for iOS). This can be done via the admin interface.

> While not required, if a verification server is required, additional

Reviewer: this is only optional in a theoretical sense. If you don't do this, the publish endpoint is wide open unless some other type of access control is applied.

Author: Yep. may be a good idea to make that clear in this. Will be elaborating and adding some details on admin interface here as well.

---
layout: default
---

# Key Publishing Guide

This page explains how to configure the server and application to allow

> publishing keys.

Reviewer 2: It might be helpful to have a short sentence or two before this. What is a "key" (there are many "keys" in this system)? What does one look like? What does it contain?

Author: I have expanded the titles of both publish and retrieval. Also going to disambiguate keys.

## Prerequisites

If you haven't started developing a mobile app, please take a look at the

> [Android Exposure Notifications Reference App](https://github.com/google/exposure-notifications-android)

Reviewer 2: Period at EOS

> You will need to

Reviewer 2: Can you write this in the present tense (e.g. "You need to" instead of "You will need to")

[complete the necessary steps](https://support.google.com/googleplay/android-developer/contact/expo_notif_api)
to use the exposure notifcations API. You should also ensure that you have
taken the steps to comply with the
[Terms of Service](https://blog.google/documents/72/Exposure_Notifications_Service_Additional_Terms.pdf).

> For instance, your app may not request any of the following permissions:

Reviewer 2: I think we should drop this lest risk getting out of sync with future updates.

Author: dropped the outlined list.  Also moved to a shared doc component.

  * ACCESS_FINE_LOCATION
  * ACCESS_COARSE_LOCATION
  * ACCESS_BACKGROUND_LOCATION
  * BLUETOOTH_ADMIN

## Publishing Keys

In order for your application to publish keys to the server, the server

> requires the addition of the Application Name (for Android) and the Bundle ID

Reviewer 2: ```suggestion
requires the registration of the Application Name (for Android) or the Bundle ID
```

(for iOS). This can be done via the admin interface.

> To add these entries, we can use the Admin Console Application. For

Reviewer 2: You noted the admin interface at the end of the last paragraph. Maybe drop that last sentence?

> To add these entries, we can use the Admin Console Application. For

Reviewer 2: s/we can use/use

instructions on how to launch the console, you can view the
[deployment documentation](/getting-started/deploying#configuring-the-server).

Once the Admin Console is launched, you will see a few choices presented.

> ![](../images/admin_console_landing.png)

Reviewer 2: Please add titles here for screen readers and accessibility.

In the Admin Console, select "Create new Authorized Application"

![](../images/admin_console_add_authorized_application.png)

> You are required to provide the App Package Name that will be publishing to

Reviewer: note that this is App Package Name and/or Bundle ID?

> You are required to provide the App Package Name that will be publishing to

Reviewer 2: It might be better to break this section out into bullets?

Author: I tried that locally and wasn't sure I liked how it rendered. Here is how this looks as in this pr.

![image](https://user-images.githubusercontent.com/638797/84177785-38705d80-aa38-11ea-869e-7a71f630ee6a.png)

the server. You can optionally configure a limited set of regions that will be
accepted. This region is included as part of the publish payload. While it is
expected this is likely a Country, State, or Provice Code, the string is opaque

> to the server. You can use this to delineate any boundary you find useful. 

Reviewer: Expand to say that the regions used on upload are used in export configuration. and that exports can be comprised of 1 or more region codes.
